### PR TITLE
Support reserved instance pricing

### PR DESCRIPTION
**Pull request by malteo**

### CODE REVIEW

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/HourlyCloudPricingDTO.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.cloudesire.platform.apiclient.dto.model.dto;
+
+import io.swagger.annotations.ApiModel;
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+@ApiModel( "Configure pricing according to order minimum duration" )
+public class HourlyCloudPricingDTO extends BaseEntityDTO
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/HourlyCloudPricingDTO.java]: MINOR Override the "equals" method in this class. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS2160)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/HourlyCloudPricingDTO.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package com.cloudesire.platform.apiclient.dto.model.dto;
+
+import io.swagger.annotations.ApiModel;
+
+import java.math.BigDecimal;
+import java.util.Map;
+import java.util.Objects;
+
+@ApiModel( description = "Configure pricing according to order minimum duration" )
```
Comment (gionn) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/HourlyCloudPricingDTO.java]:
```suggestion
@ApiModel( description = "Configure instance pricing according to order minimum duration" )
```

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/HourlyCloudPricingDTO.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        this.upfront = upfront;
+        this.price = price;
+    }
+
+    public HourlyCloudPricingDTO( @NotNull BigDecimal price )
+    {
+        this( new BigDecimal( "0.00" ), price );
+    }
+
+    public HourlyCloudPricingDTO()
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/HourlyCloudPricingDTO.java]: MINOR "upfront" is marked "javax.validation.constraints.NotNull" but is not initialized in this constructor. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS2637)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/HourlyCloudPricingDTO.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import java.math.BigDecimal;
+import java.util.Objects;
+
+public class HourlyCloudPricingDTO extends DTO
+{
+    @ApiModelProperty( "One-time upfront cost" )
+    @NotNull
+    private BigDecimal upfront;
+
+    @ApiModelProperty( "Monthly price" )
```
Comment (gionn) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/HourlyCloudPricingDTO.java]: Hourly?

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/ReservedCloudPricingDTO.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        this.upfront = upfront;
+        this.price = price;
+    }
+
+    public ReservedCloudPricingDTO( @NotNull BigDecimal price )
+    {
+        this( new BigDecimal( "0.00" ), price );
+    }
+
+    public ReservedCloudPricingDTO()
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/ReservedCloudPricingDTO.java]: MINOR "upfront" is marked "javax.validation.constraints.NotNull" but is not initialized in this constructor. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS2637)

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/CloudProviderDTO.java [hunk 6]
```diff
@@ -145,11 +148,19 @@ public void setSupportedOperatingSystems( List<OSType> supportedOperatingSystems
...
     public Map<Integer, BigDecimal> getDiscounts()
     {
         return discounts;
     }
 
+    /**
+     * @deprecated use {@link InstancePricingDTO#setReservedCloudPricing(Map)}
+     */
+    @Deprecated
     public void setDiscounts( Map<Integer, BigDecimal> discounts )
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/CloudProviderDTO.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

File: cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/CloudProviderDTO.java [hunk 7]
```diff
@@ -145,11 +148,19 @@ public void setSupportedOperatingSystems( List<OSType> supportedOperatingSystems
         this.supportedOperatingSystems = supportedOperatingSystems;
     }
 
+    /**
+     * @deprecated use {@link InstancePricingDTO#getReservedCloudPricing()}
+     */
+    @Deprecated
     public Map<Integer, BigDecimal> getDiscounts()
```
Comment (AndreaVecchi) [inline cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/CloudProviderDTO.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS1133)

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by AndreaVecchi** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**